### PR TITLE
Feature/gh 145

### PR DESCRIPTION
**Pull request by gammonpeter**

### CODE REVIEW

**Review by pjmonks — CHANGES_REQUESTED** [review]
A lot of effort went into this, good work. I'm liking how the duplication of code is reducing now. 👍 

I have found a number of issues which I would like reviewed before I approve. The more significant issues are functional, I found a few bugs with editing/saving the values - maybe the appeared because of the complex rebase. The main bugs I found are centered around:

1. Validation - either some validation isn't run, or validation prevents saving but doesn't always show error messages for the user. Examples I've found are particularly around multiplicity controls.
2. Re-editing - edit the default profile and saving does re-render the read-only view. But editing the model again without a full refresh does not load the updated state back into the dialog.

File: src/app/classification/classification.component.ts [hunk 1]
```diff
@@ -205,24 +182,21 @@ export class ClassificationComponent
...
+  save(saveItems: Array<DefaultProfileItem>) {
 
     const resource = {
-      id: this.result.id,
-      description: this.editableForm.description
+      id: this.result.id
     };
 
+    saveItems.forEach((item: DefaultProfileItem) => {
+      resource[item.displayName.toLocaleLowerCase()] = item.value;
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/classification/classification.component.ts:192]: updated to use a property name

File: src/app/classification/classification.component.ts [hunk 2]
```diff
@@ -145,24 +129,16 @@ export class ClassificationComponent
...
-  onCancelEdit() {
-    this.editMode = false; // Use Input editor whe adding a new folder.
-  }
 
   classifierDetails(id: any) {
     this.resourcesService.classifier
       .get(id)
       .subscribe((response: ClassifierDetailResponse) => {
         this.result = response.body;
+        this.result.domainType = CatalogueItemDomainType.Classification;
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/classification/classification.component.ts:138]: No when i checked the result the domain isn't present

File: src/app/shared/default-profile/default-profile.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { Component, Input, OnInit } from '@angular/core';
+
+import { DefaultProfileControls } from '@mdm/model/defaultProfileModel';
+
+@Component({
+  selector: 'mdm-default-profile',
+  templateUrl: './default-profile.component.html'
+})
+export class DefaultProfileComponent implements OnInit {
+  @Input() catalogueItem: any;
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/shared/default-profile/default-profile.component.ts:29]: i tried but a classifiers caused an issue so switched to any not idea i know

File: src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      this.dataTypeErrors = '';
+      this.dataTypeErrors =
+        'Please fill in all required values for the new Data Type';
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  fetchDataTypes = (text, loadAll, offset, limit) => {
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.ts:146]: this is being used another component so needs to be property syntax to allow it use editing service

File: src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.html [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+          <input
+            *ngIf="item.controlType === profileControlType.text"
+            matInput
+            type="text"
+            name="{item.displayName}}"
+            [(ngModel)]="item.value"
+            value="item.value"
+            class="outlined-input form-control"
+          />
+          <div *ngIf="item.controlType === profileControlType.multiplicity">
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.html:50]: added TODO in ts file

File: src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (pjmonks) [reply src/app/modals/default-profile-editor-modal/default-profile-editor-modal.component.ts:146]: OK, no problem with getting it to fit with existing code.

File: src/app/shared/default-profile/default-profile.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pjmonks) [reply src/app/shared/default-profile/default-profile.component.ts:29]: OK, if it works for now it's something that could be looked at later.

**Review by pjmonks — DISMISSED** [review]
These current changes look good to me.

Found an issue during testing, the Edit Default Profile dialog can't open when requested. Also need to fix the Jenkins issues to get that passing.

**Review by pjmonks — APPROVED** [review]
I think you've nailed it 🥳

**Pull request merged**